### PR TITLE
Add !alert and !role commands

### PR DESCRIPTION
**Pull request by conorwastakenwastaken**
# Alerts system

This PR will allow us to enable "alerts" per-thread. If you enable an alert on a thread, you'll continuously get mentioned for each message sent into the thread, until you disable it (by running the same command once more). If you schedule a thread to close, you won't get mentioned twice, it'll filter you out of the `alert_users` array since the bot will mention you either way.

# Ability to impersonate roles

Anyone with the inboxAdminRoleId can use `!role` to change what role they want to appear as. This will stay this way until they either reset it, or of course the thread is closed (so it won't follow the user to the next thread). Please note that the hard-coded ID is there for a reason, let me know if you feel there's a better way to allow that group of users to use the command!

# Misc Changes
- Changed the fall-back role name from "Moderator" to "Staff" [here](https://github.com/FlexLabs/modmailbot/blob/bdf487e8ad6bccf049f5147a826c15b87810a39b/src/data/Thread.js#L51), to make it consistent with the other fall-back role name in the same file, please let me know if it was like this for a reason, but I couldn't find any reason for it to be like that (bdf487e)
- The bot won't try and send the "Closing thread" message anymore if it has to auto-close due to the channel being deleted (be884ad)
- The `purge` and `undo` commands will send an error if no messages were found (5395f10)
- Fixed the `logs delete` command not returning when ran, resulting in both a success message and an error message (c4992ea)
- Fixed a heck ton of random whitespaces (12078db)

### CODE REVIEW

File: db/migrations/add_alert_users.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  await knex.schema.table("threads", (table) => {
+    table.string("alert_users", 20).nullable().defaultTo(null).after("scheduled_close_name");
+  });
+};
+
+exports.down = async (knex) => {
+  await knex.schema.table("threads", table => {
+    table.dropColumn("alert_users");
+  });
+};
```
Comment (bsian03) [inline db/migrations/add_alert_users.js]:
```suggestion
};

```

File: db/migrations/add_staff_role_overrides.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  await knex.schema.table("threads", (table) => {
+    table.json("staff_role_overrides").nullable().defaultTo(null).after("alert_users");
+  });
+};
+
+exports.down = async (knex) => {
+  await knex.schema.table("threads", table => {
+    table.dropColumn("staff_role_overrides");
+  });
+};
```
Comment (bsian03) [inline db/migrations/add_staff_role_overrides.js]:
```suggestion
};

```

File: db/migrations/add_alert_users.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+exports.up = async (knex) => {
+  await knex.schema.table("threads", (table) => {
+    table.string("alert_users", 20).nullable().defaultTo(null).after("scheduled_close_name");
```
Comment (bsian03) [inline db/migrations/add_alert_users.js:3]: Can't we just have an array of strings? That seems easier to handle

File: src/modules/role.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+const utils = require("../utils");
+const threadUtils = require("../threadUtils");
+
+/**
+ * @param {Eris.CommandClient} bot
+ */
+module.exports = bot => {
+  threadUtils.addInboxServerCommand(bot, "role", async (msg, args, thread) => {
+    if (! thread) return;
+    if (! args[0] || ! config.inboxAdminRoleId || (! msg.member.roles.includes(config.inboxAdminRoleId) && ! msg.member.roles.includes("523021576128692239"))) {
```
Comment (bsian03) [inline src/modules/role.js]: I feel like this massive condition could be cleaned up

**Review by bsian03 — CHANGES_REQUESTED** [review]

File: src/modules/role.js (hunk 4 above)
Comment (conorwastakenwastaken) [reply src/modules/role.js]: Updated, lmk if that's better.

**Review by salesawy1 — APPROVED** [review]

**Pull request merged**